src/Storages/Statistics/CountMinSketchStatistics.cpp

```diff
@@ -0,0 +1,100 @@
+#include <Storages/Statistics/CountMinSketchStatistics.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+#if USE_DATASKETCHES
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+}
+
+
+CountMinSketchStatistics::CountMinSketchStatistics(const SingleStatisticsDescription & stat_, DataTypePtr data_type_)
+    : IStatistics(stat_)
+    , sketch(HASH_COUNT, BUCKET_COUNT)
+    , data_type(data_type_)
+{
+}
+
+Float64 CountMinSketchStatistics::estimateEqual(const Field & value) const
+{
+    if (auto float_val = IStatistics::getFloat64(value))
+        return sketch.get_estimate(&float_val.value(), 8);
+    if (auto string_val = IStatistics::getString(value))
+        return sketch.get_estimate(string_val->data(), string_val->size());
+    UNREACHABLE();
+}
+
+bool CountMinSketchStatistics::checkType(const Field & f)
+{
+    switch (f.getType())
+    {
+        case Field::Types::Int64:
```

Comment:
The function name is too generic (l. 34), and it is not clear how UInt64, Int64, Float64 and String are related (such that true is returned for all of them). The reader wonders if these are supported types? If so, what about 8/16/32 bit Integers, Float32 and FixedString?